tracer/src/Datadog.Trace/DataStreamsMonitoring/DataStreamsContextPropagator.cs

```diff
@@ -31,7 +36,23 @@ public void Inject<TCarrier>(PathwayContext context, TCarrier headers)
     {
         if (headers is null) { ThrowHelper.ThrowArgumentNullException(nameof(headers)); }
 
-        headers.Add(DataStreamsPropagationHeaders.PropagationKey, PathwayContextEncoder.Encode(context));
+        var encodedBytes = PathwayContextEncoder.Encode(context);
+
+        int base64Length = ((encodedBytes.Length + 2) / 3) * 4;
+        byte[] base64EncodedContextBytes = new byte[base64Length];
+        var status = Base64.EncodeToUtf8(encodedBytes, base64EncodedContextBytes, out _, out int bytesWritten);
+
+        if (status != OperationStatus.Done)
+        {
+            Log.Error("Failed to encode Data Streams context to Base64. OperationStatus: {Status}", status);
```

Comment:
I think you want to bailed out here after logging the error. You were throwing an exception before, which I suggested you remove, but should you add a `return` now so the code below doesn't try to add the headers?

Reply:
Ah, good catch, thank you. Just added the return.

Reply:
Forget to push the `return` maybe as I'm not seeing it?